### PR TITLE
parenx & osmnx methods & data; evaluation WIP

### PR DESCRIPTION
**Pull request by anastassiavybornova**
this PR contains

- updated methods notebook for osmnx: node consolidation with default param of 10m, then dropping multiedges)
- updated methods notebook for parenx incl. terminal output of cli run
- data (results for all 6 fuas) for osmnx and parenx
- added read-in functions to utils for sgeop and parenx
- adjusted parenx folder structure (to mirror rest of output folders, we now have 2 folders: `parenx-voronoi` and `parenx-skeletonize`, which each contain `{fua}.parquet` file)
- lastly: WIP evaluation notebook in `evaluation`

this resolves:

- #130 
- #194 

& xref (wip) #200

### CODE REVIEW

**Comment by anastassiavybornova** [comment]
question: we now have a lot of different read-in funcs in utils, only one of those using `.explode()` on the gdf before returning it, would it make sense to have this as standard (cause it won't hurt?? or will it?) for all read-in funcs, or is there a specific reason why we only added it for the `utils.read_manual()` one?

**Comment by jGaboardi** [comment]
> question: we now have a lot of different read-in funcs in utils, only one of those using `.explode()` on the gdf before returning it, would it make sense to have this as standard (cause it won't hurt?? or will it?) for all read-in funcs, or is there a specific reason why we only added it for the `utils.read_manual()` one?

God question...

**Comment by jGaboardi** [comment]
> would it make sense to have this as standard

I think it would surely make sense to have fewer lines of code. And probably very doable with some minor refactoring.

File: core/utils.py [hunk 1]
```diff
@@ -88,18 +88,31 @@ def read_manual(fua: int, proj_crs: str | int | pyproj.CRS) -> geopandas.GeoData
...
 def read_parenx(
-    fua: int, option: str, proj_crs: str | int | pyproj.CRS
+    fua: int, proj_crs: str | int | pyproj.CRS, option: str
 ) -> geopandas.GeoDataFrame:
     """Read in prepared parenx data."""
 
     return (
-        geopandas.read_parquet(_fua_path(fua, "parenx", option=option))
+        geopandas.read_parquet(_fua_path(fua, f"parenx-{option}"))
         .explode(ingore_index=True, index_parts=False)
```
Comment (jGaboardi) [inline core/utils.py:105]: So do we need this `explode` for `parenex` read-ins?

File: core/utils.py [hunk 2]
```diff
@@ -88,18 +88,31 @@ def read_manual(fua: int, proj_crs: str | int | pyproj.CRS) -> geopandas.GeoData
...
         .to_crs(proj_crs)
     )
 
 
+def read_sgeop(fua: int, proj_crs: str | int | pyproj.CRS) -> geopandas.GeoDataFrame:
```
Comment (jGaboardi) [inline core/utils.py:110]: Should go ahead and rename to `neatnet` here? (and in data directories?)

File: notebooks/archive/evaluate_h3cells.ipynb
Comment (jGaboardi) [inline notebooks/archive/evaluate_h3cells.ipynb:1]: If we are archiving this version and keeping an updated notebook of the same name, perhaps let's append a date label --> `evaluate_h3cells_2024-12-02.ipynb`

**Review by jGaboardi — COMMENTED** [review]
To resolve [the failures](https://github.com/uscuni/simplification/actions/runs/12124401459/job/33802051167?pr=201#step:7:721) you'll need to update [`test_utils.test_read_parenex()`](https://github.com/uscuni/simplification/blob/c3e04c36eb5049f45a56d1227d0310dbd9c8e803/core/tests/test_utils.py#L61-L73).

File: core/utils.py (hunk 2 above)
Comment (anastassiavybornova) [reply core/utils.py:110]: updating this PR now,  & can take care of this renaming part

File: notebooks/archive/evaluate_h3cells.ipynb
Comment (anastassiavybornova) [reply notebooks/archive/evaluate_h3cells.ipynb:1]: agreed, will update

File: core/utils.py (hunk 1 above)
Comment (anastassiavybornova) [reply core/utils.py:105]: @jGaboardi ok just discussed with @martinfleis , consensus is to add `.explode()` in all read in funcs, will do that now

File: core/utils.py (hunk 2 above)
Comment (anastassiavybornova) [reply core/utils.py:110]: @jGaboardi taking this back :D martin will take care of the renaming in both repos at one go (later on) so sticking with `sgeop` for now

File: notebooks/archive/evaluate_h3cells.ipynb
Comment (anastassiavybornova) [reply notebooks/archive/evaluate_h3cells.ipynb:1]: ✅

**Comment by anastassiavybornova** [comment]
@jGaboardi i've updated the check, lmk what you think! the parenx_voronoi and parenx_skeletonize ones can probably be merged (DRY), i was just not sure how to pass 2 zipped and 1 non-zipped parameter to the pytest.mark.parametrize

**Review by martinfleis — APPROVED** [review]

**Review by jGaboardi — APPROVED** [review]

**Pull request merged**